src/net_processing.cpp

```diff
@@ -1302,7 +1312,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             pfrom->cleanSubVer = cleanSubVer;
         }
         pfrom->nStartingHeight = nStartingHeight;
-        pfrom->fClient = !(nServices & NODE_NETWORK);
+        pfrom->fClient = (!(nServices & NODE_NETWORK) && !(nServices & NODE_NETWORK_LIMITED));
```

Comment:
I'd rather not use fClient as a proxy for NODE_NETWORK_LIMITED. It deserves its own classification.

Reply:
Yes, this needs its own bool so that fClient gets checked when testing for our initial headers-sync peer. Then, fPreferredDownload is only set to false if we're still in IBD.

That way, we request from this peer as if they were a regular full node as long as we're synced.

Reply:
IMO we want pruned peers for header sync... but right, we need a protection to not request blocks from limited peers during IBD.
What about adding an `fLimitedNode` bool and check it at the point we populate vGetData with block requests?

Seems also simpler and more constant (NODE_NETWORK_LIMITED is not really a "client").

I changed this PR and applied my approach which I have written above.